extensions/amp-form/0.1/amp-form.js

```diff
@@ -436,29 +447,63 @@ export class AmpForm {
    */
   handleXhrSubmit_(varSubsFields, trust) {
     this.setState_(FormState_.SUBMITTING);
-
-    const p = this.doVarSubs_(varSubsFields)
-        .then(() => {
-          this.triggerFormSubmitInAnalytics_('amp-form-submit');
-          this.actions_.trigger(
-              this.form_, 'submit', /* event */ null, trust);
-          // After variable substitution
-          const values = this.getFormAsObject_();
-          // At the form submitting state, we want to display any template
-          // messages with the submitting attribute.
-          this.renderTemplate_(values);
-        })
-        .then(() => this.doActionXhr_())
-        .then(response => this.handleXhrSubmitSuccess_(response),
-            error => {
-              return this.handleXhrSubmitFailure_(/** @type {!Error} */(error));
-            });
-
+    const varSubPromise = this.doVarSubs_(varSubsFields);
+    let p;
+    if (this.ssrTemplateHelper_.isSupported()) {
+      p = varSubPromise.then(() => {
+        this.actions_.trigger(
+            this.form_, 'submit', /* event */ null, trust);
+        // Note that we do not render templates for the submitting state
+        // and only deal with submit-success or submit-error.
+        return this.ssrTemplateHelper_.fetchAndRenderTemplate(this.form_);
+      }).then(
+          resp => this.handleSsrTemplateSuccess_(resp),
+          error => this.handleSsrTemplateFailure_(/** @type {!JsonObject} */ (error)));
+    } else {
+      p = varSubPromise
+          .then(() => {
+            this.submittingWithTrust_(trust);
+            return this.doActionXhr_();
+          })
+          .then(response => this.handleXhrSubmitSuccess_(
+              /* !../../../src/service/xhr-impl.FetchResponse */ response),
```

Comment:
removed. It was there because I was exploring trying to reuse the same callback for xhr and ssr calls but the JS annotation required and es lint errors got messy, so i just created two separate callbacks to handle both cases. thanks